diff --git a/CMakeLists.txt b/CMakeLists.txt
index d7c9b752..6d43bf5b 100644
--- a/CMakeLists.txt
+++ b/CMakeLists.txt
@@ -4,6 +4,8 @@ project(autoware_behavior_velocity_planner_common)
 find_package(autoware_cmake REQUIRED)
 autoware_package()
 
+find_package(tl-expected REQUIRED)
+
 ament_auto_add_library(${PROJECT_NAME} SHARED
   src/scene_module_interface.cpp
   src/planner_data.cpp
@@ -16,6 +18,10 @@ ament_auto_add_library(${PROJECT_NAME} SHARED
   src/experimental/scene_module_interface.cpp
 )
 
+target_link_libraries(${PROJECT_NAME}
+  tl::expected
+)
+
 if(BUILD_TESTING)
   file(GLOB TEST_SOURCES test/src/*.cpp)
   ament_add_ros_isolated_gtest(test_${PROJECT_NAME} ${TEST_SOURCES})
